src/System.Text.RegularExpressions/tests/RegexParserTests.cs

```diff
@@ -0,0 +1,1076 @@
+        [InlineData(@"""(?im-x+n)""", RegexOptions.None, null, true)]
+        [InlineData(@"""(?i) """, RegexOptions.None, null, true)]
+        [InlineData(@"""(?x) """, RegexOptions.None, null, true)]
+        [InlineData(@""" (?x) """, RegexOptions.None, null, true)]
+        [InlineData(@""" (?-x) """, RegexOptions.IgnorePatternWhitespace, null, true)]
+        [InlineData(@""" ( (?-x) ) """, RegexOptions.IgnorePatternWhitespace, null, true)]
+        [InlineData(@""" (?-x:) """, RegexOptions.IgnorePatternWhitespace, null, true)]
+        [InlineData(@""" (?-x: ) """, RegexOptions.IgnorePatternWhitespace, null, true)]
+        [InlineData(@""" (?-x: (?+x: ) ) """, RegexOptions.IgnorePatternWhitespace, null, true)]
+        [InlineData(@"""(?-x""", RegexOptions.IgnorePatternWhitespace, RegexParseError.UnrecognizedGrouping, true)]
+        [InlineData(@"""(?-x """, RegexOptions.IgnorePatternWhitespace, RegexParseError.UnrecognizedGrouping, true)]
+        [InlineData(@"""(?-x :""", RegexOptions.IgnorePatternWhitespace, RegexParseError.UnrecognizedGrouping, true)]
+        [InlineData(@"""(?-x )""", RegexOptions.IgnorePatternWhitespace, RegexParseError.UnrecognizedGrouping, true)]
+        [InlineData(@"""(?-x :)""", RegexOptions.IgnorePatternWhitespace, RegexParseError.UnrecognizedGrouping, true)]
+        [InlineData(@""")""", RegexOptions.None, RegexParseError.TooManyParentheses, true)]
+        [InlineData(@"""a""", RegexOptions.None, null, true)]
+        [InlineData(@"""ab""", RegexOptions.None, null, true)]
+        [InlineData(@"""a*""", RegexOptions.None, null, true)]
+        [InlineData(@"""a*?""", RegexOptions.None, null, true)]
+        [InlineData(@"""a+""", RegexOptions.None, null, true)]
+        [InlineData(@"""a+?""", RegexOptions.None, null, true)]
+        [InlineData(@"""a?""", RegexOptions.None, null, true)]
+        [InlineData(@"""a??""", RegexOptions.None, null, true)]
+        [InlineData(@"""()""", RegexOptions.None, null, true)]
+        [InlineData(@"""(a)""", RegexOptions.None, null, true)]
+        [InlineData(@"""(""", RegexOptions.None, RegexParseError.NotEnoughParentheses, true)]
+        [InlineData(@"""(a""", RegexOptions.None, RegexParseError.NotEnoughParentheses, true)]
+        [InlineData(@"""|""", RegexOptions.None, null, true)]
+        [InlineData(@""" |""", RegexOptions.None, null, true)]
+        [InlineData(@"""| """, RegexOptions.None, null, true)]
+        [InlineData(@""" | """, RegexOptions.None, null, true)]
+        [InlineData(@"""|""", RegexOptions.IgnorePatternWhitespace, null, true)]
+        [InlineData(@""" |""", RegexOptions.IgnorePatternWhitespace, null, true)]
+        [InlineData(@"""| """, RegexOptions.IgnorePatternWhitespace, null, true)]
+        [InlineData(@""" | """, RegexOptions.IgnorePatternWhitespace, null, true)]
+        [InlineData(@"""||""", RegexOptions.None, null, true)]
+        [InlineData(@"""(|)""", RegexOptions.None, null, true)]
```

Comment:
@CyrusNajmabadi  Should the quotes from these strings also be trimmed? i.e. `@"""(|)"""` is `"(|)"` but should probably just be `(|)`?

Reply:
yes.  the test should literally be:  ```[InlineData(@"(|)", ...]```